### PR TITLE
Improve test speed

### PR DESCRIPTION
**Pull request by jonathanmetzman**
1. Allow running tests in parallel. Locally this takes the time for running all tests (including integration tests) from 6 minutes to ~50 seconds. We don't do parallel by default since it doesn't really save any time unless running integration tests on my machine (probably due to overhead of starting ~70 processes). This also speeds up CI from about 12 minutes to 6 minutes  (since github actions has 2 cores per machine).
2. Fix how we run tests. I'm not exactly sure why, but the method we used for discovering tests, recursing through every directory and passing to unittest caused the build/infra tests to execute twice. Fixing this makes running unittests take ~20 seconds instead of ~35.

This change also uses pytest for running tests since it's easy to use it to run tests in parallel.
This change was made possible by #5113

### CODE REVIEW

File: infra/ci/requirements.txt [hunk 1]
```diff
@@ -4,3 +4,5 @@ pyfakefs==4.1.0
 pylint==2.5.3
 PyYAML==5.3.1
 yapf==0.30.0
+pytest==6.2.1
+pytest-xdist==2.2.0
```
Comment (inferno-chromium) [inline infra/ci/requirements.txt:8]: nit: alpha order

File: infra/presubmit.py [hunk 2]
```diff
@@ -345,32 +340,49 @@ def get_changed_files():
...
-  for relevant_dir in relevant_dirs:
-    suite_list.append(unittest.TestLoader().discover(relevant_dir,
-                                                     pattern='*_test.py'))
-  suite = unittest.TestSuite(suite_list)
-  result = unittest.TextTestRunner().run(suite)
+  # Use ignore-glob because ignore doesn't seem to work properly with the way we
+  # pass directories to pytest.
+  command = [
+      'pytest',
+      '--ignore-glob=infra/base-images/base-sanitizer-libs-builder/*',
```
Comment (inferno-chromium) [inline infra/presubmit.py:373]: Add
"# Test errors with error: "ModuleNotFoundError: No module named 'apt'".

File: infra/pytest.ini [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[pytest]
+python_files = *_test.py
```
Comment (inferno-chromium) [inline infra/pytest.ini:2]: nit: newline.

File: infra/presubmit.py [hunk 4]
```diff
@@ -345,32 +340,49 @@ def get_changed_files():
...
+      '--ignore-glob=infra/build/*',
+  ]
+  if parallel:
+    command.extend(['-n', 'auto'])
+  command += list(relevant_dirs)
+  return subprocess.run(command, check=False).returncode == 0
```
Comment (inferno-chromium) [inline infra/presubmit.py:379]: Should we show why these failed, like print output or something.
Also, should we print something at start of function to tell "Running build tests", "Running non-build tests", etc.

**Review by inferno-chromium — APPROVED** [review]

File: infra/ci/requirements.txt (hunk 1 above)
Comment (jonathanmetzman) [reply infra/ci/requirements.txt:8]: Done.

File: infra/presubmit.py (hunk 4 above)
Comment (jonathanmetzman) [reply infra/presubmit.py:379]: >Should we show why these failed, like print output or something.

That happens already.

>Also, should we print something at start of function to tell "Running build tests", "Running non-build tests", etc.

Done

**Pull request merged**